src/java/org/apache/cassandra/transport/Frame.java

```diff
@@ -301,6 +305,10 @@ public void encode(ChannelHandlerContext ctx, Frame frame, List<Object> results)
 
             results.add(header);
             results.add(frame.body);
+
+            int messageSize = header.writerIndex() + frame.body.writerIndex();
+            ClientRequestSizeMetrics.totalBytesWritten.inc(messageSize);
```

Comment:
this is a me thing, but I would prefer this is done before `results.add`.  The code is safe, but I like assuming that once you give the stuff to netty it owns it and you shouldn't touch it anymore.